weaviate/collection/collection.py

```diff
@@ -261,12 +264,18 @@ def insert(
         uuid: Optional[UUID] = None,
         vector: Optional[List[float]] = None,
     ) -> uuid_package.UUID:
+        props = {}
+        for key, val in data.items():
+            if isinstance(val, RefToObject):
+                props[key] = val.to_beacon()
+            elif isinstance(val, datetime):
+                props[key] = val.isoformat("T") + "Z"
```

Comment:
I believe that adding a "Z" assumes the `datetime` is in UTC. However, in the absence of a user specified timezone, Python [assumes it to be local time](https://stackoverflow.com/a/23705687) (!!!) and users should be able to add their own offsets if they require dates other than UTC

[According to Weaviate](https://weaviate.io/developers/weaviate/config-refs/datatypes#datatype-date), it stores these offsets canonically and these offsets can be parsed back when constructing a datetime object upon resolution so I don't think the `+ "Z"` is required here

Reply:
Will resolve in favour of changes to dates elsewhere